packages/office-ui-fabric-react/src/components/MessageBar/MessageBar.base.tsx

```diff
@@ -122,9 +126,9 @@ export class MessageBarBase extends BaseComponent<IMessageBarProps, IMessageBarS
   }
 
   private _renderSingleLine(): React.ReactElement<React.HTMLAttributes<HTMLAreaElement>> {
-    const { theme } = this.props;
+    const theme = this._globalTheme;
     return (
-      <div style={{ background: theme!.semanticColors.bodyBackground }}>
+      <div style={{ background: theme.semanticColors.bodyBackground }}>
```

Comment:
So the fix is just wrapping the messagebar in a div whose bodybackground is set its respective value in the theme?

Reply:
Kind of, but I am having discussions with designers now to decide if we should better hard code some of the values. We want this wrapper to be always white because if we change it somehow to a darker color in a contextual theme the colors look not as intended. The whole reason for this PR is to ignore the contextual theme because the MessageBar colors have a meaning and we don't them look different regardless of the shade of the theme.

Reply:
If the wrapper is always white, then we might as well remove the transparency on the message bar backgrounds. But making it always white defeats the purpose of having it blend into the page in the list highlighting scenario.

Reply:
And that is exactly the reason why mixing these 2 usage scenarios in one slot is very tricky. On one side you would want the list highlighting blend well with theme changes and on the other side you want to preserve the meaning on the MessageBar colors.

Reply:
The list highlighting also uses colors-with-meaning. The intention there is to have just a hint of the background color of the page blended into it, so it looks more natural, same with message bar. In both cases, you should immediately recognize "this list row has an error associated with it" or "this message bar is indicating an error" from the color alone.